### PR TITLE
refactor(deps): remove stream-events dependency

### LINKED ISSUE #1916: refactor(deps): Remove `stream-events` dependency
The usage of `stream-events` can be internalized using native [streams](https://nodejs.org/api/stream.html) and [events](https://nodejs.org/api/events.html) - such as emitting `writing` on `_write` method (e.g. https://github.com/googleapis/nodejs-storage/pull/1914/commits/f666842ada519c5428a39bac12cefd01d4fcd91b#diff-eee587ca8a340f7ab2aed59173accad2fb1d776f2c176ebda172d1ea1e239f81R390-R396)

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1916  🦕

### CODE REVIEW

**Review by shaffeeullah — APPROVED** [review]
Looks straightforward enough! @danielbankhead pls take a look as well though

File: src/util.ts [hunk 1]
```diff
@@ -166,3 +167,32 @@ export function formatAsUTCISO(
...
   return resultString;
 }
+
+export class PassThroughShim extends PassThrough {
+  private shouldEmitReading = true;
+  private shouldEmitWriting = true;
+
+  constructor() {
+    super();
+  }
```
Comment (danielbankhead) [inline src/util.ts:177]: [Optional] No need for the constructor here - the [default constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor) covers this.

File: src/util.ts [hunk 2]
```diff
@@ -166,3 +167,32 @@ export function formatAsUTCISO(
...
+
+  _read(size: number): void {
+    if (this.shouldEmitReading) {
+      this.emit('reading');
+      this.shouldEmitReading = false;
+    }
+    super._read(size);
+  }
+
+  _write(
```
Comment (danielbankhead) [inline src/util.ts:187]: You'll need a `_final` handler in addition to this `_write` handler:
- https://nodejs.org/api/stream.html#writable_finalcallback

**Review by danielbankhead — COMMENTED** [review]
Looks good so far

File: src/util.ts [hunk 3]
```diff
@@ -166,3 +167,32 @@ export function formatAsUTCISO(
 
   return resultString;
 }
+
+export class PassThroughShim extends PassThrough {
+  private shouldEmitReading = true;
+  private shouldEmitWriting = true;
```
Comment (danielbankhead) [inline src/util.ts:173]: Looking at this further, 'reading' and 'writing' are called on every `_read` and `_write` event - to be fully backwards-compatible this class should do the same

Comment (ddelgrosso1) [reply src/util.ts:173]: Ah good catch, I misread the stream-events / stubs code. Will fix.

Comment (ddelgrosso1) [reply src/util.ts:173]: Actually I do believe it is only called once: https://github.com/stephenplusplus/stubs/blob/580b4d8b3a34c1193d5ec10e0db0d6ea4b4c0ec5/index.js#L30 it looks like after the number of calls is exhausted we revert to the cached function and stream-events would stop emitting reading / writing.

File: src/util.ts (hunk 2 above)
Comment (ddelgrosso1) [reply src/util.ts:183]: Added.

File: src/util.ts (hunk 1 above)
Comment (ddelgrosso1) [reply src/util.ts]: Removed

**Review by danielbankhead — APPROVED** [review]

**Pull request merged**